Podcast/SeriesDetailHeaderView.swift

```diff
@@ -64,145 +62,219 @@ class SeriesDetailHeaderView: UIView {
+        let moreTags = FillButton(type: .tag)
+        tagsView.addSubview(moreTags)
+        moreTags.setTitle("+\(series.tags.count)", for: .normal)
+        moreTags.sizeToFit()
+        moreTags.isEnabled = false
+        moreTags.addTarget(self, action: #selector(self.tagButtonPressed(button:)), for: .touchUpInside)
+        moreTags.snp.makeConstraints({ make in
+            make.leading.equalToSuperview()
+            make.centerY.equalToSuperview()
+        })
         
-        relatedTagsLabel.sizeToFit()
-        relatedTagsLabel.frame.origin.x = padding
+        // TODO: don't add a tag if it's too wide
+        var tagsNotDisplayed = 0
+        var tagsArray = [FillButton]()
+        var lastTagIndex = 0
         
-        subscribeButtonChangeState(isSelected: series.isSubscribed)
-        imageView.setImageAsynchronouslyWithDefaultImage(url: series.largeArtworkImageURL)
+        for i in 0 ..< series.tags.count {
+            let newButton = FillButton(type: .tag)
+            newButton.setTitle(series.tags[i].name, for: .normal)
+            newButton.tag = i
+            newButton.addTarget(self, action: #selector(tagButtonPressed(button:)), for: .touchUpInside)
+            newButton.isHidden = true
+            newButton.sizeToFit()
+            tagsArray.append(newButton)
+        }
         
-        var tagHeight = relatedTagsLabel.frame.maxY + 2 * marginPadding
-        if series.tags.count > 0 {
-            // Create tags (Need no tags design)
-            var remainingWidth = frame.width - 2 * padding
-            let moreTags = FillButton(type: .tag)
-            moreTags.setTitle("+\(series.tags.count)", for: .normal)
-            moreTags.sizeToFit()
-            remainingWidth = remainingWidth - (moreTags.frame.width + 2 * tagButtonInnerXPadding + tagButtonOuterXPadding)
-            var offset: CGFloat = 0
-            var numAdded = 0
-            for index in 0 ..< series.tags.count {
-                let tag = series.tags[index]
-                let tagButton = FillButton(type: .tag)
-                tagButton.setTitle(tag.name, for: .normal)
-                tagButton.sizeToFit()
-                let width = tagButton.frame.width + 2 * tagButtonInnerXPadding
+        tagsArray.sort {
+            $0.frame.width < $1.frame.width
+        }
+        
+        if tagsArray.count > 0 {
+            tagsView.addSubview(tagsArray[0])
+            tagsArray[0].snp.makeConstraints({ make in
+                make.leading.equalToSuperview()
+                make.centerY.equalToSuperview()
+                make.height.equalTo(tagButtonHeight)
+                let currWidth = tagsArray[0].frame.width + 2 * tagButtonInnerXPadding
+                make.width.equalTo(currWidth)
+            })
+            tagsArray[0].isHidden = false
+            
+            var remainingWidth = tagsView.frame.width - moreTags.frame.width - 2 * tagButtonInnerXPadding - tagButtonOuterXPadding
+            for i in 1 ..< tagsArray.count {
```

Comment:
can we just make this as tag in tagsArray ... I don't see us using the i anywhere

Reply:
I have to access the tag to the left of it to set the offset so I did `tagsArray[i-1].snp.trailing`

Reply:
ok!